### PR TITLE
Add docstring example to use dataloader_idx in transfer_batch_to_device

### LINKED ISSUE #8845: The new `dataloader_idx` argument not reflected in the doc string example of hook `transfer_batch_to_device`
https://github.com/PyTorchLightning/pytorch-lightning/blob/cb2a8ed1b8f78386de4771d3bc59e19022e1b023/pytorch_lightning/core/hooks.py#L717

A new argument dataloader_idx has been added to the hook transfer_batch_to_device. However, it is not clear in the doc string example yet how it is going to be used.

### PR DESCRIPTION
**Pull request by lsqshr**
## What does this PR do?

Fixes #8845

### Does your PR introduce any breaking changes? If yes, please list them.

no

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Yes.

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by justusschock — APPROVED** [review]

**Comment by awaelchli** [comment]
@lsqshr can you merge master here to trigger the GitHub jobs for the merge? I can't do it due to permission settings.

**Comment by lsqshr** [comment]
@awaelchli I also do not have the permission to trigger the merge. It needs to be one of the project collaborators.

**Comment by awaelchli** [comment]
I mean you merge master into this branch (or rebase it to master) using the git command locally and then push the changes. 

```bash
git pull origin master
git checkout your branch
git merge master
git push 
```
